src/main/core/support/configuration.rs

```diff
@@ -953,6 +953,89 @@ impl<'de> serde::Deserialize<'de> for ProcessArgs {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub struct Signal(nix::sys::signal::Signal);
+
+impl serde::Serialize for Signal {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.0.as_str())
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Signal {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct SignalVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for SignalVisitor {
+            type Value = Signal;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a signal string or integer")
+            }
+
+            fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_str(&v)
+            }
```

Comment:
nit: this is already the default behaviour for `visit_string`, so it's not needed

Reply:
done